pkg/storage/ingest/reader.go

```diff
@@ -281,24 +310,65 @@ func (r *PartitionReader) fetchLastCommittedOffsetWithRetries(ctx context.Contex
 	return offset, err
 }
 
-func (r *PartitionReader) fetchLastCommittedOffset(ctx context.Context) (kgo.Offset, error) {
+func (r *PartitionReader) fetchLastCommittedOffset(ctx context.Context) (int64, error) {
+	const endOffset = -1 // -1 is a special value for kafka that means "the last offset"
+
+	// We use an ephemeral client to fetch the offset and then create a new client with this offset.
+	// The reason for this is that changing the offset of an existing client requires to have used this client for fetching at least once.
+	// We don't want to do noop fetches just to warm up the client, so we create a new client instead.
 	cl, err := kgo.NewClient(kgo.SeedBrokers(r.kafkaCfg.Address))
 	if err != nil {
-		return kgo.NewOffset(), errors.Wrap(err, "unable to create admin client")
+		return endOffset, errors.Wrap(err, "unable to create admin client")
 	}
 	adm := kadm.NewClient(cl)
 	defer adm.Close()
 
 	offsets, err := adm.FetchOffsets(ctx, consumerGroup)
 	if errors.Is(err, kerr.UnknownTopicOrPartition) {
 		// In case we are booting up for the first time ever against this topic.
-		return kgo.NewOffset().AtEnd(), nil
+		return endOffset, nil
 	}
 	if err != nil {
-		return kgo.NewOffset(), errors.Wrap(err, "unable to fetch group offsets")
+		return endOffset, errors.Wrap(err, "unable to fetch group offsets")
 	}
 	offset, _ := offsets.Lookup(r.kafkaCfg.Topic, r.partitionID)
-	return kgo.NewOffset().At(offset.At), nil
+	return offset.At, nil
+}
+
+// WaitReadConsistency waits until all data produced up until now has been consumed by the reader.
+func (r *PartitionReader) WaitReadConsistency(ctx context.Context) (returnErr error) {
+	startTime := time.Now()
+	r.metrics.strongConsistencyRequests.Inc()
+
+	defer func() {
+		// Do not track failure or latency if the request was canceled (because the tracking would be falsed).
```

Comment:
```suggestion
		// Do not track failure or latency if the request was canceled (because the tracking would be incorrect).
```

"incorrect" or "misleading" sounds more intuitive